control/injector/Injector.php

```diff
@@ -585,15 +585,56 @@ public function inject($object, $asType=null) {
 		$objtype = $asType ? $asType : get_class($object);
 		$mapping = isset($this->injectMap[$objtype]) ? $this->injectMap[$objtype] : null;
 
+		$spec = empty($this->specs[$objtype]) ? array() : $this->specs[$objtype];
+
 		// first off, set any properties defined in the service specification for this
 		// object type
-		if (isset($this->specs[$objtype]) && isset($this->specs[$objtype]['properties'])) {
+		if(!empty($spec['properties']) && is_array($spec['properties'])) {
 			foreach ($this->specs[$objtype]['properties'] as $key => $value) {
 				$val = $this->convertServiceProperty($value);
 				$this->setObjectProperty($object, $key, $val);
 			}
 		}
 
+		// Populate named methods
+		if (!empty($spec['calls']) && is_array($spec['calls'])) {
+			foreach ($spec['calls'] as $method) {
+				// Ignore any blank entries from the array; these may be left in due to config system limitations
+				if(!$method) continue;
+
+				// Format validation
+				if(!is_array($method) || !isset($method[0]) || isset($method[2])) {
+					throw new \InvalidArgumentException(
+						"'calls' entries in service definition should be 1 or 2 element arrays."
+					);
+				}
+				if(!is_string($method[0])) {
+					throw new \InvalidArgumentException("1st element of a 'calls' entry should be a string");
+				}
+				if(isset($method[1]) && !is_array($method[1])) {
+					throw new \InvalidArgumentException("2nd element of a 'calls' entry should an arguments array");
+				}
+
+				// Check that the method exists
+				try {
```

Comment:
It will probably be more performant to do this (reflection is fairly expensive):

```
$objectMethod = array($object, $method[0]);
if (!is_callable($objectMethod)) {
//
}
```